### PR TITLE
Rd 55 add validation

### PR DESCRIPTION
**Pull request by SashaRoyenko**
Create new validation annotation to check equals of two fields. Also,  create ValidationProcessor to validate entity

### CODE REVIEW

File: src/main/java/com/robosh/ejournal/data/entity/admin/Admin.java [hunk 1]
```diff
@@ -39,7 +41,7 @@
     @Column(name = "last_name", nullable = false)
     private String lastName;
 
-    @Pattern(regexp = "[A-Za-z0-9+_.-]+@[a-z.-]+\\.[a-z]{2,8}")
+    @Email
```
Comment (NazarShchur) [inline src/main/java/com/robosh/ejournal/data/entity/admin/Admin.java:44]: nice

File: src/main/java/com/robosh/ejournal/exception/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.robosh.ejournal.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
```
Comment (NazarShchur) [inline src/main/java/com/robosh/ejournal/exception/GlobalExceptionHandler.java]: empty line

File: src/main/java/com/robosh/ejournal/service/AdminService.java [hunk 3]
```diff
@@ -18,14 +18,15 @@
...
     private final AdminMapper adminMapper;
 
-    public AdminInfoDto save(UpdateAdminDto updateAdminDto) {
-        if (!updateAdminDto.getPassword().equals(updateAdminDto.getConfirmedPassword())) {
-            throw new ValidationException("Password should be same");
-        }
-        Admin admin = adminMapper.fromUpdateAdminDtoToAdmin(updateAdminDto);
-        if (updateAdminDto.getId() == null) {
+    public AdminInfoDto save(SaveAdminDto saveAdminDto) {
+
```
Comment (NazarShchur) [inline src/main/java/com/robosh/ejournal/service/AdminService.java]: empty line

File: src/main/java/com/robosh/ejournal/util/validation/ValidatorProcessor.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        ValidatorFactory validatorFactory = Validation.buildDefaultValidatorFactory();
+        validator = validatorFactory.usingContext().getValidator();
+    }
+
+    private ValidatorProcessor() {}
+
+    @SneakyThrows
+    public static void validate(Object object) {
+        Set<ConstraintViolation<Object>> validates = validator.validate(object);
+
```
Comment (NazarShchur) [inline src/main/java/com/robosh/ejournal/util/validation/ValidatorProcessor.java:34]: empty line

File: src/main/java/com/robosh/ejournal/data/entity/admin/Admin.java (hunk 1 above)
Comment (SashaRoyenko) [reply src/main/java/com/robosh/ejournal/data/entity/admin/Admin.java:44]: Thanks)

**Review by nikitabortnichuk — APPROVED** [review]
Good!

**Review by NazarShchur — APPROVED** [review]

**Pull request merged**